src/shc.c

```diff
@@ -622,14 +628,15 @@ static const char * RTC[] = {
 "#endif",
 "		close(0);",
 "		mine = !open(proc, O_RDWR|O_EXCL);",
-"		if (!mine && errno != EBUSY)",
-"			mine = !ptrace(PT_ATTACHEXC, pid, 0, 0);",
-"		if (mine) {",
-"			kill(pid, SIGCONT);",
-"		} else {",
-/*"			fprintf(stderr, \"%s is being traced!\\n\", argv0);",*/
+"		if (!mine && errno != EBUSY) {",
+"			if((mine=2*!ptrace(PT_ATTACHEXC, pid, 0, 0)))wait(0);",
```

Comment:
you are using assignment here, is that intended?

Reply:
Yes, `mine` must assume value 2 if the ptrache() succeeds